skvlite/__init__.py

```diff
@@ -158,65 +150,71 @@ def remove(self, key: K) -> None:
-        """Return an iterator over the keys in the dictionary."""
         return self.keys()
 
-    def keys(self) -> Generator[K, None, None]:  # type: ignore[override]
-        """Return an iterator over the keys in the dictionary."""
+    def keys(self) -> Generator[K, None, None]:
         for row in self._exec_sql("SELECT key_value FROM dict ORDER BY rowid"):
             yield pickle.loads(row[0])[0]
 
-    def values(self) -> Generator[V, None, None]:  # type: ignore[override]
-        """Return an iterator over the values in the dictionary."""
+    def values(self) -> Generator[V, None, None]:
         for row in self._exec_sql("SELECT key_value FROM dict ORDER BY rowid"):
             yield pickle.loads(row[0])[1]
 
-    def items(self) -> Generator[Tuple[K, V], None, None]:  # type: ignore[override]
-        """Return an iterator over the items in the dictionary."""
+    def items(self) -> Generator[Tuple[K, V], None, None]:
         for row in self._exec_sql("SELECT key_value FROM dict ORDER BY rowid"):
             yield pickle.loads(row[0])
 
     def nbytes(self) -> int:
-        """Return the size of the dictionary in bytes."""
         return cast(int,
                     next(self._exec_sql("SELECT page_size * page_count FROM "
                                         "pragma_page_size(), pragma_page_count()")
                          )[0]
                     )
 
     def __repr__(self) -> str:
-        """Return a string representation of the dictionary."""
         return f"{type(self).__name__}({self.filename}, nitems={len(self)})"
 
     def clear(self) -> None:
-        """Remove all entries from the dictionary."""
         self._exec_sql("DELETE FROM dict")
 
     def store_if_not_present(self, key: Any, value: Any) -> None:
         self.store(key, value, _skip_if_present=True)
 
-    def vacuum(self) -> None:
+    def vacuum_and_report_size(self) -> Tuple[int, int]:
+        """Perform VACUUM operation and return size before and after."""
+        uncompressed_size = os.path.getsize(self.filename)
         self._exec_sql("VACUUM")
+        self.conn.commit()
+        compressed_size = os.path.getsize(self.filename)
+        return uncompressed_size, compressed_size
 
     def close(self) -> None:
         self.conn.close()
 
+    def enable_zstd_compression(self, table: str, column: str, compression_level: int = 19, dict_chooser: str = "''a''") -> None:
+        """Enable zstd compression for a specific table and column."""
+        # Create the table if not exists
+        self._exec_sql(
+            f"CREATE TABLE IF NOT EXISTS {table} (id INTEGER PRIMARY KEY AUTOINCREMENT, {column} TEXT)"
+        )
+
+        compression_config = f'{{"table": "{table}", "column": "{column}", "compression_level": {compression_level}, "dict_chooser": "{dict_chooser}"}}'
+        self.conn.execute(f"SELECT zstd_enable_transparent('{compression_config}')")
+        self.conn.execute('SELECT zstd_incremental_maintenance(null, 1)')
```

Comment:
It looks like the user needs to manually enable the compression. Instead, could you make it automatic (as part of `KVStore.__init__`)?